docs/CppInteropability/GettingStartedWithC++Interop.md

```diff
@@ -0,0 +1,217 @@
+#  Getting started with C++ Interoperability
+
+This document is desgined to get you started with bidirectional API-level interoperability between Swift and C++.
+
+## Table of Contents
+
+- [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code)
+- [Adding C++ to an Xcode project](#adding-c-to-an-xcode-project)
+- [Creating a Swift Package](#Creating-a-Swift-Package)
+- [Building with CMake](#building-with-cmake)
+
+## Creating a Module to contain your C++ source code
+
+- Create a new C++ implementation and header file
+- For this example we will call the files Cxx, so we should have a Cxx.cpp and Cxx.hpp.
+- Create a module.modulemap file to expose your source code 
+- In order to expose our C++ code we must designate a separate as the source.
+- Create an empty file and call it `module.modulemap`
+- In this file create the module for your source code, and define your C++ header (`requires cplusplus` isn't required but it's convention for C++ modules, especially if they use C++ features).
+```
+//In module.modulemap
+module Cxx {
+    //note that your header should be the file that containts your method implementations
+    header "Cxx.cpp
+    requires cplusplus
+}
+```
+- Move the newly created files (Cxx.cpp, Cxx.hpp, module.modulemap) into a separate directory (this should remain in your project directory)
+
+<img width="252" alt="Screen Shot 2022-02-26 at 9 14 06 PM" src="https://user-images.githubusercontent.com/62521716/155867937-9d9d6c62-4418-414d-bc4e-5d12c2055022.png">
+
+## Adding C++ to an Xcode project
+- In your xcode project, follow the steps [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code) in your project directory
+
+Add the C++ module as a Swift Compiler flag in your project's directory
+- Navigate to your project directory 
+- In the`Targets` navigate to `Build Settings` -> `Swift Compiler - Custom Flags`
+- Under `Other Swift Flags` add your search path to the C++ module with the `-I` prefix to import
+
+```
+//Add to Other Swift Flags
+-I./ProjectName/Cxx
+```
+
+- In the`Project` navigate to `Build Settings` -> `Swift Compiler - Custom Flags`
+- - Under `Other Swift Flags` add your search path to the C++ module with the `-I` prefix to import, and `-Xfrontend -enable-cxx-interop` 
```

Comment:
This is good for enabling interop, but we can actually use the import paths section instead of `-I`. So you can say something like:
- In the`Project` navigate to `Build Settings` -> `Swift Compiler - Search Paths`
- Under `Import Paths` add your search path to the C++ module (i.e, `./ProjectName/Cxx`).

Reply:
Cool cool